dask/typing.py

```diff
@@ -149,6 +157,171 @@ def __dask_keys__(self) -> NestedKeys:
         """
         raise NotImplementedError("Inheriting class must implement this method.")
 
+
+@runtime_checkable
+class NewDaskCollection(Protocol):
```

Comment:
I'm currently still experimenting. For now, I'm introducing a new collections protocol and am implementing branches for either collection. This causes a bit of duplication but feels nicer than something hybrid.

I will clean this up once we converge on a final API